data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/EventTypeCreatorImpl.java

```diff
@@ -0,0 +1,95 @@
+import io.fabric8.kubernetes.client.dsl.Resource;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import org.apache.commons.codec.binary.Hex;
+
+public class EventTypeCreatorImpl implements EventTypeCreator {
+
+    private static final Integer DNS1123_SUBDOMAIN_MAX_LENGTH = 253;
+
+    private final MixedOperation<EventType, KubernetesResourceList<EventType>, Resource<EventType>> eventTypeClient;
+
+    private MessageDigest messageDigest;
+
+    public EventTypeCreatorImpl(KubernetesClient kubernetesClient) {
+        this.eventTypeClient = kubernetesClient.resources(EventType.class);
+        try {
+            this.messageDigest = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException ignored) {
+            this.messageDigest = null;
+        }
+    }
+
+    private String getName(CloudEvent event, DataPlaneContract.Reference reference) {
+        final var suffixString = event.getType() + event.getSource() + reference.getNamespace() + reference.getName();
+        this.messageDigest.reset();
+        this.messageDigest.update(suffixString.getBytes());
+        final var suffix = Hex.encodeHexString(this.messageDigest.digest());
+        final var name = String.format("et-%s-%s", reference.getName(), suffix).toLowerCase();
+        if (name.length() > DNS1123_SUBDOMAIN_MAX_LENGTH) {
+            return name.substring(0, DNS1123_SUBDOMAIN_MAX_LENGTH);
+        }
+        return name;
+    }
+
+    private boolean eventTypeExists(String etName, DataPlaneContract.Reference reference) {
+        var et = this.eventTypeClient
+                .inNamespace(reference.getNamespace())
+                .withName(etName)
+                .get();
+        return et != null;
+    }
+
+    @Override
+    public void create(CloudEvent event, DataPlaneContract.Reference ownerReference) {
+        if (this.messageDigest == null) {
+            return;
+        }
+
+        var name = this.getName(event, ownerReference);
+        if (this.eventTypeExists(name, ownerReference)) {
+            return;
+        }
+
+        var et = new EventTypeBuilder()
+                .withReference(KReference.fromDataPlaneReference(ownerReference))
+                .withOwnerReference(new OwnerReferenceBuilder()
+                        .withName(ownerReference.getName())
+                        .withKind(ownerReference.getKind())
+                        .withApiVersion(ownerReference.getGroupVersion())
+                        .withUid(ownerReference.getUuid())
+                        .build())
+                .withNamespace(ownerReference.getNamespace())
+                .withName(name)
+                .withType(event.getType())
+                .withSource(event.getSource())
+                .withSchema(event.getDataSchema())
+                .withDescription("Event Type auto-created by controller");
+
+        this.eventTypeClient.resource(et.build()).create();
```

Comment:
we should avoid failing when the resource exist (like when it is created by some other process between checking exists and this line)

Reply:
We need to be sure on what happens when already exists and handle that case accordingly

Reply:
I looked on their github and they said that the methods generally throw exceptions, so I'll add a try-catch to handle that case